wrangler-service/src/main/java/co/cask/wrangler/service/directive/DirectivesService.java

```diff
@@ -672,6 +673,9 @@ public void execute(HttpServiceRequest request, HttpServiceResponder responder,
     } catch (DirectiveParseException e) {
       LOG.warn(e.getMessage(), e);
       error(responder, e.getMessage());
+    } catch (RecipeException e) {
+      LOG.error(e.getMessage(), e);
+      error(responder, e.getCause().getMessage());
```

Comment:
In general, we should avoid unwrapping exceptions like this, as context is lost. For example, if somebody threw a new RecipeException(msg, cause), this would lose the actual message they explicitly set. This also assumes a cause is always present, which does not have to be the case.

If the exception class is getting into the message, it probably because something is throwing new RecipeException(cause) without specifying a message. In that scenario, Throwable will toString() the cause to generate the message. 

All that to say, I suspect the actual problem is that someplace `throw new RecipeException(e)` is being called. If you change it to `throw new RecipeException(e.getMessage(), e)`, I believe it will fix the exception class in the message.